src/covid19model/visualization/optimization.py

```diff
@@ -0,0 +1,146 @@
+
+    """Plot model fit to user provided data 
+
+    Parameters
+    -----------
+    model: model object
+        correctly initialised model to be fitted to the dataset
+    data: array
+        list containing dataseries
+    startDate: string, format DD-MM-YYY
+        date corresponding to first entry of dataseries
+    states: array
+        list containg the names of the model states that correspond to the data
+    checkpoints : dict, optional
+        A dictionary with a "time" key and additional parameter keys,in the form of
+        ``{"time": [t1, t2, ..], "param": [param1, param2, ..], ..}``
+        indicating new parameter values at the corresponding timestamps.
+    samples: dict, optional
+        A dictionary containing parameter values obtained from a sampling algorithm (f.i. MCMC)
+    filename: string, optional
+        Filename + extension to save a copy of the plot_fit
+    ax : matplotlib.axes.Axes, optional
+        If provided, will use the axis to add the lines.
+
+    Returns
+    -----------
+
+    Notes
+    -----------
+
+    Example use
+    -----------
+
+
+    """
+
+    # Initialize figure and visualize data
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # check if ax object is provided by user
+    if ax is None:
+        fig, ax = plt.subplots()
+    # Create shifted index vector using self.extraTime
+    idx = pd.date_range(startDate,freq='D',periods=data[0].size + model.extraTime) - datetime.timedelta(days=model.extraTime)
+    # Plot data
+    for i in range(len(data)):
+        ax.scatter(idx[model.extraTime:],data[i],color="black")
+
+    # Perform simulation(s) and plot model prediction
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # Compute number of dataseries
+    n = len(data)
+    # Compute simulation time
+    T = data[0].size+model.extraTime-1
+    # Perform simulation(s)
+    if samples is None:
+        k = 0
+        while k < 200:
+            out = model.sim(T,checkpoints=checkpoints)
+            out = out.sum(dim="stratification")
+            for i in range(len(data)):
+                data2plot = out[states[i]].to_array(dim="states").values.ravel()
+                ax.plot(idx,data2plot,linewidth=0.25,alpha=0.2,color="blue")
+            k = k +1
+    else:
+        original_parameters = model.parameters.copy()
+        k=0
+        while k < 100:
+            for key,value in samples:
+                # do random draw and assign to model
+                model.parameters[key] = random.choice(value)
+            # run simulation
+            out = model.sim(T,checkpoints=checkpoints)
+            out = out.sum(dim="stratification")
+            for i in range(len(data)):
+                data2plot = out[states[i]].to_array(dim="states").values.ravel()
+                ax.plot(idx,data2plot)
+            k = k+1
+        # reset parameters
+        model.parameters=original_parameters
```

Comment:
Why do we add the model simulations itself inside this function? It makes the modules more interdependent (some would call it _spaghetti code_ ;-)) instead of so-called _orthogonal_ (not hard interlinked) with each function just a single purpose and action: for this function: make a scatter plot from a data set and a model output (that's it, no side-effects or doing things other functions should do, like doing the actual simulation).

Reply:
This issue has been resolved. The function now takes a list with dataseries, the startdate and the model output as input arguments and returns the ax object. To plot multiple 'runs' on top of each other I use the following code:

`
ax = plot_fit(y_model,data,start_date,lag_time,states,plt_kwargs={'linewidth': 0.25,'alpha': 0.2})

for i in range(100):
    y_model = model.sim(int(T))
    ax = plot_fit(y_model,data,start_date,lag_time,states,ax=ax,plt_kwargs={'linewidth': 0.25,'alpha': 0.2})
`
I'm unsure on how to remove the lag_time component from this function..